### PR TITLE
mangal: fix build; modernize

### LINKED ISSUE #328283: Build failure: PACKAGENAME
### Steps To Reproduce

1. build mangal

### Build log

```
mangal> evaling implicit 'postInstall' string hook
mangal> panic: panic: panic: mkdir /homeless-shelter: permission deniedmkdir /homeless-shelter: permission denied
mangal> mkdir /homeless-shelter: permission deniedgoroutine
mangal> 1
mangal> goroutine  [
mangal> 1runninggoroutine  [1]:
mangal>  [runningrunning]:
mangal> ]:
mangal> github.com/samber/lo.mustgithub.com/samber/lo.mustgithub.com/samber/lo.must((({{0xed91e00xed91e0, , {0xc0003be0f00xc0003be0f00xed91e0}}, , , {{0xc0003be0f00x00x0}, , , 0x00x0{, , 0x00x00x0, }}0x0)
mangal> ,       )
mangal> github.com/samber/lo@v1.37.0/errors.go0x0       :}github.com/samber/lo@v1.37.0/errors.go53)
mangal> : +     0x1df53github.com/samber/lo@v1.37.0/errors.go
mangal>  +:0x1df53
mangal>  +github.com/samber/lo.Must00x1df(
mangal> ...github.com/samber/lo.Must0)
mangal> github.com/samber/lo.Must0(     (...github.com/samber/lo@v1.37.0/errors.go...)
mangal> :)
mangal> 72
mangal> github.com/samber/lo@v1.37.0/errors.gogithub.com/samber/lo@v1.37.0/errors.go:github.com/metafates/mangal/where.mkdir72:(
mangal> 72github.com/metafates/mangal/where.mkdir
mangal> ({0xc000043500{, 0xc0000434e00x1f, }0x1f)
mangal> }       )
mangal> github.com/metafates/mangal/where/where.go      :github.com/metafates/mangal/where/where.go18: +github.com/metafates/mangal/where.mkdir180x4f +
mangal> (0x4fgithub.com/metafates/mangal/where.Cache
mangal> ()
mangal> github.com/metafates/mangal/where.Cache ()
mangal>         {github.com/metafates/mangal/where/where.go:88 +0xaa
mangal> github.com/metafates/mangal/where.Queries()
mangal> 0xc000043500    github.com/metafates/mangal/where/where.go, :0x1f55} +)
mangal> 0x13
mangal> github.com/metafates/mangal/where/where.go:github.com/metafates/mangal/query.init18( +github.com/metafates/mangal/where/where.go)
mangal> :0x4f88
mangal>  +github.com/metafates/mangal/query/query.go0xaa:github.com/metafates/mangal/where.Cache16
mangal> ( +github.com/metafates/mangal/where.Queries)
mangal> 0x17(
mangal> )
mangal> github.com/metafates/mangal/where/where.go      :github.com/metafates/mangal/where/where.go88: +550xaa +
mangal> 0x13github.com/metafates/mangal/where.Queries
mangal> ()
mangal> github.com/metafates/mangal/query.init  (github.com/metafates/mangal/where/where.go)
mangal> :       55github.com/metafates/mangal/query/query.go +:0x1316
mangal>  +github.com/metafates/mangal/query.init0x17(
mangal> )
mangal>         github.com/metafates/mangal/query/query.go:16 +0x17
```

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.0, NixOS, 24.11 (Vicuna), 24.11.20240716.9355fa8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix but for lesbians) 2.91.0-devpre20240717_6856720
System type: x86_64-linux
Additional system types:
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/isabel/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/isabel/.local/share/flatpak/exports/etc/xdg/nix/nix.conf:/var/lib/flatpak/exports/etc/xdg/nix/nix.conf:/home/isabel/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/isabel/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/isabel/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/cjw4phhd13ql3pn5bjgqn6032z6m8c3x-lix-2.91.0-devpre20240717_6856720/share`
 - nixpkgs: `/nix/store/77ijmdbv96rhqj2sr7aq0i5q8p16ii0d-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by isabelroses**
## Description of changes

closes https://github.com/NixOS/nixpkgs/issues/328283

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ma/mangal/package.nix [hunk 1]
```diff
@@ -30,8 +38,7 @@ buildGoModule rec {
   doCheck = false; # test fail because of sandbox
 
   meta = with lib; {
-    description =
-      "A fancy CLI app written in Go which scrapes, downloads and packs manga into different formats";
+    description = "A fancy CLI app written in Go which scrapes, downloads and packs manga into different formats";
```
Comment (eclairevoyant) [inline pkgs/by-name/ma/mangal/package.nix:41]:
```suggestion
    description = "CLI app written in Go which scrapes, downloads and packs manga into different formats";
```
removes article and subjective statement as per https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

**Review by eclairevoyant — COMMENTED** [review]
Additionally, please add the formatting commit to [`.git-blame-ignore-revs`](https://github.com/NixOS/nixpkgs/blob/master/.git-blame-ignore-revs)

**Comment by bertof** [comment]
Ok for the updates to metadata and formatting. The build phase is ok as well, but since the code from the source is not changed this seems like a regression of either Nix or `buildGoModule`, as if the environment variable is not passed to the command. The `MANGAL_CONFIG_PATH` is supposed to tell mangal where to create the config file, with a fallback to `/homeless-shelter`. Still, creating and setting $HOME is a fine solution.

**Review by bertof — APPROVED** [review]

**Review by bertof — APPROVED** [review]

**Comment by teto** [comment]
Merging since maintainer is ok

**Pull request merged**